### PR TITLE
Support partition key

### PR DESCRIPTION
**Pull request by xige-16**
use case
```
from pymilvus import CollectionSchema, FieldSchema, DataType

book_id = FieldSchema(name="book_id", dtype=DataType.INT64, is_primary=True)

book_name = FieldSchema(name="book_name", dtype=DataType.VARCHAR, max_length=200, is_partition_key=true)

book_intro = FieldSchema(name="book_intro", dtype=DataType.FLOAT_VECTOR, dim=2)

default_schema = CollectionSchema(fields=[book_id, book_name, book_intro],description="Test book search")

collection = Collection(name="hello_milvus", schema=default_schema, shards_num=1, partitions=100)
```

### CODE REVIEW

File: pymilvus/client/abstract.py [hunk 1]
```diff
@@ -175,6 +180,7 @@ def dict(self):
             "collection_id": self.collection_id,
             "consistency_level": self.consistency_level,
             "properties": self.properties,
+            "partitions": self.partitions,
```
Comment (XuanYang-cn) [inline pymilvus/client/abstract.py:183]: please change "partitions" into "num_partitions"

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Comment by longjiquan** [comment]
Please remember to cherry-pick this feature to master branch. @xige-16

**Comment by xige-16** [comment]
https://github.com/milvus-io/pymilvus/issues/1458

**Pull request merged**